Models/BaseModel.php

```diff
@@ -36,4 +40,37 @@ public function getUuid(): string
     {
         return uuid();
     }
+
+    /**
+     * function for saving image to data/files/$folderName/$fileName directory
+     *
+     * @param string $folderName
+     * @param string $filename
+     * @return bool
+     */
+    public function saveImage(string $folderName, string $filename): bool
+    {
+        $fileDir = base_path('data/files/' . $folderName . '/');
+
+        if (!is_dir($fileDir)) {
+            mkdir($fileDir, 0755, true);
+        }
+
+        $allowedFormats = ['jpg', 'jpeg', 'png', 'gif'];
+
+        $uploadedFile = $_FILES['productFile'];
+        $fileExtension = strtolower(pathinfo($uploadedFile['name'], PATHINFO_EXTENSION));
+
+        if (!in_array($fileExtension, $allowedFormats)) {
```

Comment:
Move file extension validation on the top of method

Reply:
File extension validation has been moved to the top of the method